media-libs/openimageio/openimageio-2.4.12.0.ebuild

```diff
@@ -0,0 +1,187 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+FONT_PN=OpenImageIO
+PYTHON_COMPAT=( python3_{9..11} )
+
+TEST_OIIO_IMAGE_COMMIT="aae37a54e31c0e719edcec852994d052ecf6541e"
+TEST_OEXR_IMAGE_COMMIT="df16e765fee28a947244657cae3251959ae63c00"
+inherit cmake font python-single-r1
+
+DESCRIPTION="A library for reading and writing images"
+HOMEPAGE="https://sites.google.com/site/openimageio/ https://github.com/OpenImageIO"
+SRC_URI="https://github.com/OpenImageIO/oiio/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+SRC_URI+=" test? (
+		https://github.com/OpenImageIO/oiio-images/archive/${TEST_OIIO_IMAGE_COMMIT}.tar.gz -> ${PN}-oiio-test-image-${TEST_OIIO_IMAGE_COMMIT}.tar.gz
+		https://github.com/AcademySoftwareFoundation/openexr-images/archive/${TEST_OEXR_IMAGE_COMMIT}.tar.gz -> ${PN}-oexr-test-image-${TEST_OEXR_IMAGE_COMMIT}.tar.gz
+	)"
+S="${WORKDIR}/oiio-${PV}"
+
+LICENSE="BSD"
+SLOT="0/$(ver_cut 1-2)"
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc64 ~riscv"
+
+X86_CPU_FEATURES=(
+	aes:aes sse2:sse2 sse3:sse3 ssse3:ssse3 sse4_1:sse4.1 sse4_2:sse4.2
+	avx:avx avx2:avx2 avx512f:avx512f f16c:f16c
+)
+CPU_FEATURES=( ${X86_CPU_FEATURES[@]/#/cpu_flags_x86_} )
+
+IUSE="dicom doc ffmpeg gif iv jpeg2k opencv openvdb ptex python raw test +truetype ${CPU_FEATURES[@]%:*}"
+REQUIRED_USE="python? ( ${PYTHON_REQUIRED_USE} )"
+
+# Not quite working yet
+RESTRICT="!test? ( test )" # test"
+
+BDEPEND="
+	doc? (
+		app-doc/doxygen
+		dev-texlive/texlive-bibtexextra
+		dev-texlive/texlive-fontsextra
+		dev-texlive/texlive-fontutils
+		dev-texlive/texlive-latex
+		dev-texlive/texlive-latexextra
+	)
+"
+RDEPEND="
+	dev-libs/boost:=
+	dev-cpp/robin-map
+	dev-libs/libfmt:=
+	dev-libs/pugixml:=
+	>=media-libs/libheif-1.7.0:=
+	media-libs/libjpeg-turbo:=
+	media-libs/libpng:0=
+	>=media-libs/libwebp-0.2.1:=
+	>=dev-libs/imath-3.1.2-r4:=
+	>=media-libs/opencolorio-2.1.1-r4:=
+	>=media-libs/openexr-3:0=
+	media-libs/tiff:=
+	sys-libs/zlib:=
+	dicom? ( sci-libs/dcmtk )
+	ffmpeg? ( media-video/ffmpeg:= )
+	gif? ( media-libs/giflib:0= )
+	iv? (
+		media-libs/glew:=
+		virtual/glu
+		virtual/opengl
+		|| (
+			dev-qt/qtbase:6[gui,widgets,opengl]
```

Comment:
As per qt policy we are going to need to add a `qt6` flag for toggling this: https://wiki.gentoo.org/wiki/Project:Qt/Policies#Handling_different_versions_of_Qt

Reply:
Also there needs to be some switch in the configure phase to actually instruct the build system to use qt5 or qt6. 

The user can have both installed at the same time and we can't let the build system automagically decide which to use.

Reply:
From https://github.com/OpenImageIO/oiio/blob/v2.4.12.0/INSTALL.md?plain=1#L33 :
```md
### Optional dependencies -- features may be disabled if not found
* If you are building the iv viewer (which will be disabled if any of these are not found):
     * Qt5 >= 5.6 (tested through 5.15) or Qt6 (tested through 6.4)
     * OpenGL
```

https://github.com/OpenImageIO/oiio/blob/v2.4.12.0/src/cmake/externalpackages.cmake#L252:
```cmake
# Qt -- used for iv
option (USE_QT "Use Qt if found" ON)
if (USE_QT)
    checked_find_package (OpenGL)   # used for iv
endif ()
if (USE_QT AND OPENGL_FOUND)
    checked_find_package (Qt6 COMPONENTS Core Gui Widgets OpenGLWidgets)
    if (NOT Qt6_FOUND)
        checked_find_package (Qt5 COMPONENTS Core Gui Widgets OpenGL)
    endif ()
    if (NOT Qt5_FOUND AND NOT Qt6_FOUND AND APPLE)
        message (STATUS "  If you think you installed qt with Homebrew and it still doesn't work,")
        message (STATUS "  try:   export PATH=/usr/local/opt/qt/bin:$PATH")
    endif ()
endif ()
´´´
So OpenGL is required for and only for `iv`.

It wouldn't care if you set `qt5` or `qt6`. If you know a way of making it adhere to that let me know.

There are `USE_OPENGL` and `USE_QT` but that is only relevant for `iv`. So you could leave the `qt5`, `qt6` and `opengl` flags in, but that is just a very convoluted way of saying build `iv`.
And to make sure `qt5` builds against Qt5 you would have to add blockers against `dev-qt/qtbase:6`, I am not sure if that is good design, that's why I ended up with this.

Reply:
`cmake_use_find_package` from `cmake.eclass` can be used to prevent the build system from finding an installed package if the use flag is not enabled.

https://devmanual.gentoo.org/eclass-reference/cmake.eclass/index.html

Reply:
That seems like way to much work, to fix the qt detection that should be fixed upstream and would have gone unnoticed had I not looked into the source that much. Considering this is done to help fix the fact that this currently does not build due to:
```
CMake Error at /usr/lib64/cmake/OpenColorIO/OpenColorIOTargets.cmake:70 (set_target_properties):
  The link interface of target "OpenColorIO::OpenColorIO" contains:
 
    expat::expat
 
  but the target was not found.  Possible reasons include:
 
    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.
 
Call Stack (most recent call first):
  /usr/lib64/cmake/OpenColorIO/OpenColorIOConfig.cmake:22 (include)
  src/cmake/modules/FindOpenColorIO.cmake:31 (find_package)
  src/cmake/checked_find_package.cmake:127 (find_package)
  src/cmake/externalpackages.cmake:177 (checked_find_package)
  CMakeLists.txt:155 (include)
```

So we could either mask the `iv` USE-flag or I add:
```sh
USE=+="qt5 qt6"
REQUIRED_USE="^^ ( qt5 qt6 )"
BDEPEND+="qt6? ( !dev-qt/qtbase:6 )"
```
seeing that `iv` is a tool not the main feature for `openimageio`.